var/spack/repos/builtin/packages/gaudi/package.py

```diff
@@ -146,6 +151,10 @@ def cmake_args(self):
             # todo:
             self.define("GAUDI_USE_INTELAMPLIFIER", False),
         ]
+        # Release notes for v39.0: https://gitlab.cern.ch/gaudi/Gaudi/-/releases/v39r0
+        # Gaudi@39: needs C++ >= 20, and we need to force CMake to use C++ 20 with old gcc:
+        if self.spec.satisfies("@39: %gcc@:13"):
```

Comment:
This I didn't understand, when compiling with GCC 14 C++20 will also be needed, right?

Reply:
I think starting with gcc13 the default would be c++20. However, I am not sure if `GAUDI_CXX_STANDARD` interferes with that(?).

Reply:
As far as my little, older experience with cmake goes, cmake is eager to tell g++ which C++ standard to use (I think this depends on which compiler version it detects also). At least in my case, it did already pass -std=c++17, and I did not find where that would have been set in Gaudi.

Take this with a grain of salt as what I saw is not defintive.

But yes, I think it's safer to pin the C++ standard choice from spack than to leave it to the cmake defaults algorithm is or whatever the compiler's default is because we know that in this case, Gaudi-39+ needs C++20.